src/com/xilinx/rapidwright/edif/EDIFTools.java

```diff
@@ -544,7 +546,18 @@ public static void connectPortInstsThruHier(EDIFHierPortInst src, EDIFHierPortIn
 	    // Disconnect sink from existing net if connected
 	    EDIFNet snkNet = finalSnk.getNet();
 	    if(snkNet != null) {
-	        snkNet.removePortInst(finalSnk.getPortInst());
+	        if(createdSrcNet && snkNet.getParentCell() == finalSrc.getNet().getParentCell()) {
+	            // Let's delete the net we created and use the existing snkNet instead
+	            EDIFNet net = finalSrc.getNet();
+	            if(snkNet != net) {
+	                net.getParentCell().removeNet(net);
+	                snkNet.addPortInst(finalSrc.getPortInst());
+	            } else {
+	                return;
+	            }
```

Comment:
```suggestion
	            }
	            return;
```
Is it safe to return immediately because at this point `finalSrc.getNet() == finalSnk.getNet()` and thus `finalSrc.getNet().addPortInst(finalSnk.getPortInst());` is not necessary?